### PR TITLE
20200928 Fixes - 40mms, Vendors, Weapon Fixes

### LINKED ISSUE #2815: Armor kits and you: Apparently the Covenant's underfunded
The UNSC currently get 600 normal repair kits, and 400 miniature repair kits on the ODP
As well as a further 600 normal repair kits and 400 miniature repair kits in the KS7 base
The URF have 400 of each (not counting the extra included in the SoE armory thanks to their limited membership numbers)

The sangheili posses
30 regular repair kits
24 miniature repair kits
The jiralhanae posses
20 regular repair kits
16 miniature repair kits
The unggoy posses 
24 normal repair kits
32 miniature repair kits
The yanme'e posses
16 normal repair kits
20 miniature repair kits (especially notable as they cannot remove their armor to repair)
The kig-yar posses
36 armor repair kits only.

This equals 126 regular repair kits
92 miniature repair kits

This is quite imbalanced, as UNSC and even URF posses several times more repair kits than the covenant, which is even worse when one considers that the repair kit are locked by race, instead of the massive pool of repair kits that everyone gets access too.
(Especially worse when you consider UNSC have far more repair kits than Covenant and URF combined)
Server Revision: 78153a58b7dad251b8ec3738453d2bd1210f19d5 - alpha -

### PR DESCRIPTION
**Pull request by 5crownik007**
Closes #2815 
🆑 5crownik007
tweak: 40mm available for purchase in cargo shuttle. landmines are available too
tweak: 40mm illumination shells now illuminate point of impact
tweak: ODST vendors now vend M6S mags instead of M6D mags
bugfix: body scanners now function
tweak: covenant now have ample supply of armourkits
bugfix: ma3 fixed finally
tweak: unsc plasma weapons have damage parity with their ballistic counterparts
/🆑

### CODE REVIEW

File: code/modules/halo/weapons/ammo_g40mm.dm [hunk 1]
```diff
@@ -36,6 +36,15 @@
...
 	projectile_type = /obj/item/projectile/bullet/g40mm/illumination
 
+/obj/item/device/flashlight/flare/g40mm
+	icon = 'code/modules/halo/weapons/icons/Weapon Sprites.dmi'
+	name = "illumination shell"
+	brightness_on = 7
+	light_color = "#ffff99"
+	icon_state = "40mm_shell_illumination"
+	item_state = null
+	on = 1
```
Comment (X0-11) [inline code/modules/halo/weapons/ammo_g40mm.dm]: better to make the code that spawns this flare activate it normally

Comment (5crownik007) [reply code/modules/halo/weapons/ammo_g40mm.dm]: why? This works just fine from testing

Comment (5crownik007) [reply code/modules/halo/weapons/ammo_g40mm.dm]: Eh there are probably unintended consequences I didn't account for you're right

Comment (X0-11) [reply code/modules/halo/weapons/ammo_g40mm.dm]: flarecode activation may have special handling effects of it's own or have them added later. Utilising the generalised method of activation would be better

Comment (5crownik007) [reply code/modules/halo/weapons/ammo_g40mm.dm]: done

File: code/modules/halo/unsc/supply/ammo.dm [hunk 2]
```diff
@@ -141,14 +141,40 @@
...
 /decl/hierarchy/supply_pack/unsc_ammo/m19_spanker
 	name = "M19 SPNKr rockets (1 box of 3 dual tubes)"
 	cost = 1200
 	contains = list(/obj/item/weapon/storage/box/spnkr = 1)
 	containername = "\improper M19 SPNKr crate"
 
+/* M301 */
+
+/decl/hierarchy/supply_pack/unsc_ammo/g40mm_he
+	name = "M301 40mm HE crate(1 box)"
```
Comment (bloxgate) [inline code/modules/halo/unsc/supply/ammo.dm]: Personally, I'd add a space before the `(1 box)`

File: code/modules/halo/weapons/unsc_plasma.dm [hunk 3]
```diff
@@ -133,7 +133,8 @@
 /obj/item/projectile/energy/unsc_plasma_light
 	icon = 'code/modules/halo/weapons/icons/Weapon Sprites.dmi'
 	icon_state = "plasma_proj_light"
-	damage = 25
+	damage = 30
+	shield_damage = -5
```
Comment (bloxgate) [inline code/modules/halo/weapons/unsc_plasma.dm:137]: Is this supposed to be negative?

File: code/modules/halo/covenant/structures_machines/covvendors.dm [hunk 4]
```diff
@@ -145,8 +145,6 @@
 		/obj/item/flight_item/covenant_pack = 1,
 		/obj/item/dumb_ai_chip/cov = 2,
 		/obj/item/weapon/plastique/covenant = 8,
-		/obj/item/weapon/armor_patch/cov = 8,
-		/obj/item/weapon/armor_patch/mini/cov = 10,
```
Comment (bloxgate) [inline code/modules/halo/covenant/structures_machines/covvendors.dm:149]: Unless I'm missing something about how this works, doesn't this mean the covenant now have no armor kits, with the exception of in the Kig-Yar vendors?

I was under the impression that the issue was them having too few, not too many.

**Review by bloxgate — CHANGES_REQUESTED** [review]

File: code/modules/halo/weapons/unsc_plasma.dm (hunk 3 above)
Comment (5crownik007) [reply code/modules/halo/weapons/unsc_plasma.dm:137]: Yes

File: code/modules/halo/covenant/structures_machines/covvendors.dm (hunk 4 above)
Comment (5crownik007) [reply code/modules/halo/covenant/structures_machines/covvendors.dm:149]: pointbased vendors default to 100 in a pool iirc if amount is not set, you are missing something

File: code/modules/halo/unsc/supply/ammo.dm (hunk 2 above)
Comment (5crownik007) [reply code/modules/halo/unsc/supply/ammo.dm]: done

**Review by bloxgate — APPROVED** [review]

**Pull request merged**